### ISSUE TITLE
`pyright` thinks `slice` is not hashable while it is

### ISSUE THREAD

**Issue by bersbersbers** [microsoft/pyright#9236]
**Describe the bug**
`pyright` thinks `slice` is not hashable. But it is since 3.12, see https://github.com/python/cpython/issues/84783.

**Code or Screenshots**
```python
{slice(0)}
```

```
c:\Git\project\bug5.py
  c:\Git\project\bug5.py:1:1 - error: Expression value is unused (reportUnusedExpression)
  c:\Git\project\bug5.py:1:2 - error: Set entry must be hashable
    Type "slice" is not hashable (reportUnhashable)
2 errors, 0 warnings, 0 informations
```

**VS Code extension or command-line**
`pyright 1.1.384`

**Comment by erictraut** [comment]
According to the typeshed stubs, `slice` is [not hashable](https://github.com/python/typeshed/blob/2370b8b9d194d4a468907df8952ef0dd7daff187/stdlib/builtins.pyi#L951). If you think this is incorrect, please file an issue or a PR in the [typeshed project](https://github.com/python/typeshed/issues). 

I'm going to close this because pyright is working correctly here based on the type information it's provided.

**Comment by bersbersbers** [comment]
Thanks! I thought it must be a pyright bug because mypy does not complain, but maybe they just don't implement this check.

I opened https://github.com/python/typeshed/issues/12824.
